modules/shared/src/main/scala/org/tessellation/schema/metagraphMessage.scala

```diff
@@ -0,0 +1,20 @@
+package org.tessellation.schema
+
+import org.tessellation.schema.address.Address
+import org.tessellation.security.signature.Signed
+
+import derevo.cats.{eqv, show}
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+
+object metagraphMessage {
+  sealed trait SigningMessage {
+    def signed: Signed[Address]
+  }
+
+  @derive(eqv, show, encoder, decoder)
+  case class AdminFeeKeyMessage(signed: Signed[Address]) extends SigningMessage
+
+  @derive(eqv, show, encoder, decoder)
+  case class StakingMessage(signed: Signed[Address]) extends SigningMessage
```

Comment:
I honestly think that it should rather be these classes that get signed and also I think some ordinal would be required. So that a message can't be replayed and can't overwrite new message with an old one.

Reply:
Key note of distinction here, if you serialize them both and the data is equivalent, it's not enough, it needs some differentiator (even besides ordinal.)

Reply:
Yes with kryo these classes will be serialized as different binaries. With json though they may be serialized to the same output - hence for the json we need to modify the logic to also include the top level class to be persisted. We could also add a json classes registry used to salt the binary right beofre signing, and to be added to the binary before checking the signature.

Reply:
This should be addressed on recent change with field names